ios/src/TangramMap/TGMapViewController.h

```diff
@@ -0,0 +1,731 @@
+//
+//  TGMapViewController.h
+//  TangramiOS
+//
+//  Created by Matt Blair on 8/25/14.
+//  Updated by Matt Smollinger on 7/29/16.
+//  Copyright (c) 2016 Mapzen. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+#import <GLKit/GLKit.h>
+
+@class TGMapViewController;
+
+#import "TGGeoPoint.h"
+#import "TGGeoPolygon.h"
+#import "TGGeoPolyline.h"
+#import "TGSceneUpdate.h"
+#import "TGHttpHandler.h"
+#import "TGLabelPickResult.h"
+#import "TGMarkerPickResult.h"
+
+/**
+ Description for a <a href="https://mapzen.com/documentation/tangram/Cameras-Overview/#camera-types">
+ Tangram camera type</a>.
+ */
+typedef NS_ENUM(NSInteger, TGCameraType) {
+    /// Type for a <a href="https://mapzen.com/documentation/tangram/Cameras-Overview/#perspective-camera">perspective camera</a>
+    TGCameraTypePerspective = 0,
+    /// Type for an <a href="https://mapzen.com/documentation/tangram/Cameras-Overview/#isometric-camera">isometric camera</a>
+    TGCameraTypeIsometric,
+    /// Type for a <a href="https://mapzen.com/documentation/tangram/Cameras-Overview/#flat-camera">flat camera</a>
+    TGCameraTypeFlat,
+};
+
+/**
+ These defines describe an ease type function to be used for camera or other transition animation.
+ The function is being used to interpolate between the start and end position of the animation.
+ */
+typedef NS_ENUM(NSInteger, TGEaseType) {
+    /// Linear ease type `f(t) = t`
+    TGEaseTypeLinear = 0,
+    /// Cube ease type `f(t) = (-2 * t + 3) * t * t`
+    TGEaseTypeCubic,
+    /// Quint ease type `f(t) = (6 * t * t - 15 * t + 10) * t * t * t`
+    TGEaseTypeQuint,
+    /// Sine ease type `f(t) = 0.5 - 0.5 * cos(PI * t)`
+    TGEaseTypeSine,
+};
+
+/// Debug flags to render additional information about various map components
+typedef NS_ENUM(NSInteger, TGDebugFlag) {
+    /// While on, the set of tiles currently being drawn will not update to match the view
+    TGDebugFlagFreezeTiles = 0,
+    /// Apply a color change to every other zoom level to visualize proxy tile behavior
+    TGDebugFlagProxyColors,
+    /// Draw tile boundaries
+    TGDebugFlagTileBounds,
+    /// Draw tile infos (tile coordinates)
+    TGDebugFlagTileInfos,
+    /// Draw label bounding boxes and collision grid
+    TGDebugFlagLabels,
+    /// Draw tangram infos (framerate, debug log...)
+    TGDebugFlagTangramInfos,
+    /// Draw all labels (including labels being occluded)
+    TGDebugFlagDrawAllLabels,
+    /// Draw tangram frame graph stats
+    TGDebugFlagTangramStats,
+    /// Draw feature selection framebuffer
+    TGDebugFlagSelectionBuffer,
+};
+
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ A gesture recognizer delegate that can be implemented to receive gesture events from the map view.
```

Comment:
Just a thought - much of the documentation below refers to a "logical pixel" - it might make sense up here to define just what that is because most developers won't understand there's a fair amount of translation from OpenGL -> UIKit pixel space.